cfme/configure/settings.py

```diff
@@ -53,272 +50,339 @@ def __init__(self, description=None, scope=None, days=None, hours=None, timezone
         self.timezone = timezone
 
     def create(self, cancel=False):
-        navigate_to(self, 'Add')
-        fill(self.timeprofile_form, {'description': self.description,
-                                     'scope': self.scope,
-                                     'days': self.days,
-                                     'hours': self.hours,
-                                     'timezone': self.timezone,
-                                     })
+        view = navigate_to(self, 'Add')
+        new_timeprofile = view.timeprofile_form.fill({
+            'description': self.description,
+            'scope': self.scope,
+            'days': self.days,
+            'hours': self.hours,
+            'timezone': self.timezone,
+        })
+
         if not cancel:
-            sel.click(self.save_button)
+            view.timeprofile_form.save_button.click()
             end = "saved" if self.appliance.version > '5.7' else "added"
-            flash.assert_success_message('Time Profile "{}" was {}'
-                                         .format(self.description, end))
+            FlashMessages('Time Profile "{}" was {}'.format(self.description, end))
+        return new_timeprofile
 
     def update(self, updates):
-        navigate_to(self, 'Edit')
-        fill(self.timeprofile_form, {'description': updates.get('description'),
-                                     'scope': updates.get('scope'),
-                                     'timezone': updates.get('timezone')},
-             action={version.LOWEST: form_buttons.save,
-                     '5.7': self.save_edit_button})
-        flash.assert_success_message(
-            'Time Profile "{}" was saved'.format(updates.get('description', self.description)))
+        view = navigate_to(self, 'Edit')
+        row = view.timeprofile_form.table.row()
+        row[1].click()
+        changed = view.timeprofile_form.fill({
+            'scope': updates.get('scope'),
```

Comment:
Only `scope` can be edited?

Reply:
In the test case we are just editing the scope hence I have made a provision to edit the scope.

Reply:
What about other fields? It should be more common.

Reply:
okay will be updating the code.